tests/vcs/test_clone.py

```diff
@@ -208,3 +208,44 @@ def test_clone_unknown_subprocess_error(mocker, clone_dir):
             clone_to_dir=str(clone_dir),
             no_input=True,
         )
+
+
+@pytest.mark.parametrize(
+    'url, expected_clone_args',
+    [
+        (
+            'https://github.com/foo/bar',
+            ['git', 'clone', '--recurse-submodules', 'https://github.com/foo/bar'],
+        ),
+        (
+            'https://bitbucket.org/foo/bar',
+            ['hg', 'clone', 'https://bitbucket.org/foo/bar'],
+        ),
+    ],
+)
+def test_clone_recurse_submodules(mocker, clone_dir, url, expected_clone_args):
```

Comment:
This test too don't really check the code.
Check that the clone command is as you excepted

Reply:
That is exactly what's done in the test, we validate the clone command has the correct code for git clones and hg clones.